services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatDaoImpl.kt

```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    override fun updateHankeGeometriat(hankeGeometriat: HankeGeometriat) {
+        with(jdbcOperations) {
+            update("""
+                UPDATE HankeGeometriat
+                SET
+                    version = ?,
+                    modifiedByUserId = ?,
+                    modifiedAt = ?
+                WHERE
+                    id = ?
+            """.trimIndent()) { ps ->
+                ps.setInt(1, hankeGeometriat.version!!)
+                if (hankeGeometriat.modifiedByUserId != null) {
+                    ps.setInt(2, hankeGeometriat.modifiedByUserId!!)
+                } else {
+                    ps.setNull(2, Types.INTEGER)
+                }
+                if (hankeGeometriat.modifiedAt != null) {
+                    ps.setTimestamp(3, Timestamp(hankeGeometriat.modifiedAt!!.toInstant().toEpochMilli()))
+                } else {
+                    ps.setNull(3, Types.TIMESTAMP)
+                }
+                ps.setInt(4, hankeGeometriat.id!!)
+            }
+            // delete old geometry rows
+            deleteHankeGeometriaRows(hankeGeometriat, this)
+            // save new geometry rows
+            saveHankeGeometriaRows(hankeGeometriat, this)
+        }
+    }
+
+    override fun retrieveHankeGeometriat(hankeId: Int): HankeGeometriat? {
+        with(jdbcOperations) {
+            val hankeGeometriat = query("""
+            SELECT
+                id,
+                hankeId,
+                version,
+                createdByUserId,
+                createdAt,
+                modifiedByUserId,
+                modifiedAt
+            FROM HankeGeometriat WHERE hankeId = ?            
+        """.trimIndent(), { rs, _ ->
+                HankeGeometriat(
+                        rs.getInt(1),
+                        rs.getInt(2),
+                        null,
+                        rs.getInt(3),
+                        rs.getInt(4),
+                        rs.getTimestamp(5).toInstant().atZone(TZ_UTC),
+                        rs.getInt(6),
+                        rs.getTimestamp(7).toInstant().atZone(TZ_UTC)
+                )
+            }, hankeId).getOrNull(0)
+            return if (hankeGeometriat != null) {
+                hankeGeometriat.featureCollection = FeatureCollection()
+                hankeGeometriat.featureCollection!!.features = mutableListOf()
+                query("""
+                    SELECT
+                        ST_AsGeoJSON(geometria),
+                        parametrit
+                    FROM
+                        HankeGeometria
+                    WHERE
+                        hankeGeometriatId = ?
+                """.trimIndent(), { rs, _ ->
+                    val geojson = rs.getString(1)
+                    val paramjson = rs.getString(2)
+                    hankeGeometriat.featureCollection!!.features.add(Feature().apply {
+                        geometry = OBJECT_MAPPER.readValue(geojson)
```

Comment:
This is a bit hard to read. Could it be split into smaller (maybe even separately testable?) pieces.

Reply:
I splitted reading the HankeGeometria rows into separate function